### PR TITLE
PostgreSQL Fedora  images update

### PR DESCRIPTION
**Pull request by phracek**
This pull request fixes several issues:

PSQL16 is bump to F40,
PSQL12 and PSQL13 are not build and tested on Fedora. Fedora 38 has reached EOL already.

The same is valid also for PSQL14.

### CODE REVIEW

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
Rebased on the top of master

[test]

**Comment by phracek** [comment]
Rebased on the top of master

[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
Fix package names for Fedora 40.

[test]

**Comment by phracek** [comment]
In postgresql container is function called 'wait_ready' that checks if container is up. But if checks it forever. We should fail after sometime.

[test]

File: src/Dockerfile.fedora [hunk 1]
```diff
@@ -49,28 +49,25 @@ COPY root/usr/libexec/fix-permissions /usr/libexec/fix-permissions
...
-    INSTALL_PKGS+=" pgaudit" && \
-{% endif %}
 {% if spec.version == "16" %}
-    INSTALL_PKGS+=" postgresql16-pgaudit" && \
-{% endif %}
-{% if spec.version not in ["16"] %}
-    dnf -y module enable postgresql:{{ spec.version }} && \
+    INSTALL_PKGS+=" pgaudit" && \
+{% else %}
+    INSTALL_PKGS+=" postgresql15-pgaudit" && \
```
Comment (hhorak) [inline src/Dockerfile.fedora:61]: Same as above, `postgresql{{ spec.short }}` might be a bit more future looking.

**Review by hhorak — COMMENTED** [review]
LGTM, just one nitpick about using variable instead of hard-coding 15. Otherwise merge-able from my PoV.

File: src/Dockerfile.fedora (hunk 1 above)
Comment (phracek) [reply src/Dockerfile.fedora]: https://github.com/sclorg/postgresql-container/pull/583/commits/cb913c75bdea8a6f2ef634527e20dee1db14bf67

File: src/Dockerfile.fedora [hunk 2]
```diff
@@ -49,28 +49,25 @@ COPY root/usr/libexec/fix-permissions /usr/libexec/fix-permissions
 # safe in the future. This should *never* change, the last test is there
 # to make sure of that.
 {% if spec.version == "16" %}
-RUN INSTALL_PKGS="rsync tar gettext bind-utils postgresql16-server postgresql16-contrib nss_wrapper " && \
-    INSTALL_PKGS+=" procps-ng util-linux postgresql16-upgrade" && \
+RUN INSTALL_PKGS="rsync tar gettext bind-utils postgresql-server postgresql-contrib nss_wrapper " && \
+    INSTALL_PKGS+=" procps-ng util-linux postgresql-upgrade" && \
 {% else %}
-RUN INSTALL_PKGS="rsync tar gettext bind-utils postgresql-server postgresql-contrib nss_wrapper postgresql-upgrade procps-ng util-linux" && \
+RUN INSTALL_PKGS="rsync tar gettext bind-utils postgresql15-server postgresql15-contrib nss_wrapper postgresql15-upgrade procps-ng util-linux" && \
```
[the remark this answers is not in the document]
Comment (phracek) [reply src/Dockerfile.fedora]: https://github.com/sclorg/postgresql-container/pull/583/commits/cb913c75bdea8a6f2ef634527e20dee1db14bf67

**Comment by phracek** [comment]
[test-all]

**Pull request merged**